File: intersecting_rectangles_php/rectangle.php
<?php

class Point {
  public $x = null;
  public $y = null;

  public function __construct(int $x, int $y) {
    $this->x = $x;
    $this->y = $y;
  }
}

class Rectangle {
  public $lowerLeft;
  public $upperRight;

  public function __construct(Point $lowerLeft, Point $upperRight) {
    $this->lowerLeft = $lowerLeft;
    $this->upperRight = $upperRight;
  }

  public static function getOverlappingRectangle(Rectangle $r1, Rectangle $r2) {
    $dummyRect = new Rectangle(new Point(0, 0), new Point(1, 1))

    return $dummyRect;
  }
}

File: intersecting_rectangles/rectangle.cs
public class Point {
    public int _x;
    public int _y;

    public Point(int x, int y) {
        _x = x;
        _y = y;
    }
}

public class Rectangle {
    public Point _lowerLeft;
    public Point _upperRight;

    public Rectangle(Point lowerLeft, Point upperRight) {
        _lowerLeft = lowerLeft;
        _upperRight = upperRight;
    }

    public static getOverlappingRectangle(Rectangle r1, Rectangle r2) {
        
    }
}